In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data = pd.read_csv("https://raw.githubusercontent.com/dphi-official/Datasets/master/covid_19_data.csv")

In [3]:
data.head()

SNo ObservationDate Province/State  Country/Region      Last Update  \
0    1      01/22/2020          Anhui  Mainland China  1/22/2020 17:00   
1    2      01/22/2020        Beijing  Mainland China  1/22/2020 17:00   
2    3      01/22/2020      Chongqing  Mainland China  1/22/2020 17:00   
3    4      01/22/2020         Fujian  Mainland China  1/22/2020 17:00   
4    5      01/22/2020          Gansu  Mainland China  1/22/2020 17:00   

   Confirmed  Deaths  Recovered  
0        1.0     0.0        0.0  
1       14.0     0.0        0.0  
2        6.0     0.0        0.0  
3        1.0     0.0        0.0  
4        0.0     0.0        0.0

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40805 entries, 0 to 40804
Data columns (total 8 columns):
SNo                40805 non-null int64
ObservationDate    40805 non-null object
Province/State     22790 non-null object
Country/Region     40805 non-null object
Last Update        40805 non-null object
Confirmed          40805 non-null float64
Deaths             40805 non-null float64
Recovered          40805 non-null float64
dtypes: float64(3), int64(1), object(4)
memory usage: 2.5+ MB


In [5]:
data.describe()

SNo      Confirmed        Deaths      Recovered
count  40805.000000   40805.000000  40805.000000   40805.000000
mean   20403.000000    7403.745448    465.412621    2728.499547
std    11779.533204   26483.022808   2392.737598   16413.103519
min        1.000000       0.000000      0.000000       0.000000
25%    10202.000000      44.000000      0.000000       0.000000
50%    20403.000000     458.000000      7.000000      29.000000
75%    30604.000000    2836.000000     82.000000     570.000000
max    40805.000000  405843.000000  41128.000000  547386.000000

In [6]:
data.shape

(40805, 8)

In [7]:
data.nunique()

SNo                40805
ObservationDate      143
Province/State       709
Country/Region       223
Last Update         1884
Confirmed          10830
Deaths              3173
Recovered           5482
dtype: int64

In [8]:
data.sort_values(by = 'Confirmed', ascending=False)

SNo ObservationDate                        Province/State  \
32120  32121      05/31/2020                                   NaN   
31565  31566      05/30/2020                                   NaN   
31011  31012      05/29/2020                                   NaN   
40577  40578      06/12/2020                              New York   
39848  39849      06/11/2020                              New York   
39123  39124      06/10/2020                              New York   
38412  38413      06/09/2020                              New York   
30458  30459      05/28/2020                                   NaN   
37728  37729      06/08/2020                              New York   
37043  37044      06/07/2020                              New York   
36359  36360      06/06/2020                              New York   
35675  35676      06/05/2020                              New York   
35003  35004      06/04/2020                              New York   
34339  34340      06/03/2020                              New York   
33675  33676      06/02/2020                              New York   
33012  33013      06/01/2020                              New York   
32397  32398      05/31/2020                              New York   
30009  30010      05/27/2020                                   NaN   
31842  31843      05/30/2020                              New York   
31288  31289      05/29/2020                              New York   
30734  30735      05/28/2020                              New York   
30220  30221      05/27/2020                              New York   
29768  29769      05/26/2020                              New York   
29316  29317      05/25/2020                              New York   
29557  29558      05/26/2020                                   NaN   
28864  28865      05/24/2020                              New York   
28412  28413      05/23/2020                              New York   
27960  27961      05/22/2020                              New York   
27508  27509      05/21/2020                              New York   
27056  27057      05/20/2020                              New York   
...      ...             ...                                   ...   
7300    7301      03/20/2020          Diamond Princess cruise ship   
7301    7302      03/20/2020                         French Guiana   
7302    7303      03/20/2020                            Guadeloupe   
7303    7304      03/20/2020                               Mayotte   
7304    7305      03/20/2020                               Reunion   
7305    7306      03/20/2020                                   NaN   
7306    7307      03/20/2020                                   NaN   
7307    7308      03/20/2020                                   NaN   
7308    7309      03/20/2020                                   NaN   
7309    7310      03/20/2020                                   NaN   
7310    7311      03/20/2020                                   NaN   
28902  28903      05/24/2020                             Recovered   
28950  28951      05/24/2020                               Unknown   
28951  28952      05/24/2020                               Unknown   
7006    7007      03/19/2020                                   NaN   
2477    2478      02/26/2020  Lackland, TX (From Diamond Princess)   
7001    7002      03/19/2020          Diamond Princess cruise ship   
7002    7003      03/19/2020                         French Guiana   
7003    7004      03/19/2020                            Guadeloupe   
7004    7005      03/19/2020                               Mayotte   
7005    7006      03/19/2020                               Reunion   
7007    7008      03/19/2020                                   NaN   
28953  28954      05/24/2020                               Unknown   
7008    7009      03/19/2020                                   NaN   
7009    7010      03/19/2020                                   NaN   
7010    7011      0

Russia has the highest number of cumulative cases up to 05/31/2020 - 405843 cases.

In [12]:
country_data = data.groupby('Country/Region').sum()
country_data = country_data.drop('SNo', axis=1)
country_data

Confirmed     Deaths   Recovered
Country/Region                                                   
 Azerbaijan                            1.0        0.0         0.0
('St. Martin',)                        2.0        0.0         0.0
Afghanistan                       491513.0    10035.0     53623.0
Albania                            61819.0     2200.0     41160.0
Algeria                           406979.0    34862.0    214148.0
Andorra                            55520.0     3086.0     33136.0
Angola                              3446.0      201.0       936.0
Antigua and Barbuda                 1766.0      191.0       925.0
Argentina                         631181.0    23157.0    187677.0
Armenia                           341665.0     4925.0    125741.0
Aruba                                 19.0        0.0         0.0
Australia                         528736.0     6304.0    365376.0
Austria                          1205301.0    39822.0    881383.0
Azerbaijan                        227556.0     2754.0    132222.0
Bahamas                             5828.0      708.0      2208.0
Bahamas, The                          10.0        0.0         0.0
Bahrain                           441662.0      888.0    238673.0
Bangladesh                       1580627.0    23462.0    306530.0
Barbados                            6139.0      423.0      3438.0
Belarus                          1616491.0     9370.0    597334.0
Belgium                          3480698.0   524742.0    860372.0
Belize                              1215.0      132.0       738.0
Benin                              11178.0      140.0      4942.0
Bhutan                              1416.0        0.0       392.0
Bolivia                           314919.0    11816.0     37658.0
Bosnia and Herzegovina            137000.0     6979.0     74746.0
Botswana                            1793.0       74.0       758.0
Brazil                          17172054.0  1009660.0   7428254.0
Brunei                             11823.0       94.0      9052.0
Bulgaria                          128430.0     6361.0     40253.0
...                                    ...        ...         ...
Sweden                           1860490.0   206708.0    160063.0
Switzerland                      2238469.0   119501.0   1630983.0
Syria                               4709.0      252.0      1960.0
Taiwan                             35374.0      525.0     22515.0
Tajikistan                        105297.0     1491.0     48439.0
Tanzania                           26422.0     1040.0      8494.0
Thailand                          222645.0     3679.0    174607.0
The Bahamas                            3.0        0.0         0.0
The Gambia                             1.0        0.0         0.0
Timor-Leste                         1407.0        0.0      1022.0
Togo                               16866.0      640.0      7812.0
Trinidad and Tobago                 9074.0      589.0      5274.0
Tunisia                            70790.0     3004.0     37867.0
Turkey                           8716050.0   231054.0   5181840.0
UK                              13951022.0  1943716.0     64725.0
US                              88972448.0  4993653.0  16537321.0
Uganda                             15019.0        0.0      3784.0
Ukraine                          1011834.0    28398.0    308809.0
United Arab Emirates             1361652.0    10957.0    579527.0
Uruguay                            50439.0     1162.0     33694.0
Uzbekistan                        172580.0      721.0    115004.0
Vatican City                           4.0        0.0         0.0
Venezuela                          58426.0      855.0     15440.0
Vietnam                            24134.0        0.0     17418.0
West Bank and Gaza                 28333.0      174.0     15290.0
Western Sahara                       460.0       18.0       288.0
Yemen                              10009.0     2138.0       403.0
Zambia                             34314.0      349.0     20312.0
Zimbabw

In [13]:
country_data.sort_values(by='Confirmed', ascending=False)

Confirmed     Deaths   Recovered
Country/Region                                                   
US                              88972448.0  4993653.0  16537321.0
Brazil                          17172054.0  1009660.0   7428254.0
Spain                           16095371.0  1763203.0   8411642.0
Italy                           15872416.0  2140610.0   7104480.0
Russia                          14949259.0   161716.0   5053499.0
UK                              13951022.0  1943716.0     64725.0
France                          12238098.0  1703149.0   3719836.0
Germany                         12184741.0   466247.0   8950846.0
Mainland China                  10213288.0   463060.0   8063275.0
Turkey                           8716050.0   231054.0   5181840.0
Iran                             8547110.0   490411.0   6143799.0
India                            6549556.0   193451.0   2761326.0
Peru                             5584390.0   157122.0   2216793.0
Canada                           4509465.0   308984.0   2100945.0
Belgium                          3480698.0   524742.0    860372.0
Chile                            3352303.0    41255.0   1780251.0
Mexico                           3172638.0   345951.0   2144173.0
Saudi Arabia                     3153750.0    20550.0   1745970.0
Netherlands                      2878548.0   346268.0     14750.0
Pakistan                         2692333.0    55685.0    824307.0
Switzerland                      2238469.0   119501.0   1630983.0
Qatar                            1985185.0     1447.0    759380.0
Portugal                         1884815.0    74044.0    502460.0
Ecuador                          1877764.0   133003.0    485136.0
Sweden                           1860490.0   206708.0    160063.0
Belarus                          1616491.0     9370.0    597334.0
Bangladesh                       1580627.0    23462.0    306530.0
Singapore                        1479025.0     1257.0    600157.0
Ireland                          1421915.0    83771.0    953476.0
United Arab Emirates             1361652.0    10957.0    579527.0
...                                    ...        ...         ...
French Guiana                        117.0        0.0         6.0
Palestine                             86.0        0.0         0.0
Lesotho                               73.0        0.0        26.0
occupied Palestinian territory        25.0        0.0         0.0
Mayotte                               21.0        0.0         0.0
Republic of Ireland                   21.0        0.0         0.0
Aruba                                 19.0        0.0         0.0
Saint Barthelemy                      17.0        0.0         0.0
Faroe Islands                         10.0        0.0         0.0
Bahamas, The                          10.0        0.0         0.0
Gibraltar                              7.0        0.0         1.0
Jersey                                 6.0        0.0         0.0
Guam                                   6.0        0.0         0.0
Vatican City                           4.0        0.0         0.0
Gambia, The                            4.0        0.0         0.0
Puerto Rico                            3.0        0.0         0.0
Cayman Islands                         3.0        0.0         0.0
Guernsey                               3.0        0.0         0.0
The Bahamas                            3.0        0.0         0.0
Greenland                              3.0        0.0         0.0
Curacao                                2.0        0.0         0.0
St. Martin                             2.0        0.0         0.0
('St. Martin',)                        2.0        0.0         0.0
East Timor                             1.0        0.0         0.0
Channel Islands                        1.0        0.0         0.0
North Ireland                          1.0        0.0         0.0
Cape Verde                             1.0        0.0         0.0
The Gambia                             1.0        0.0         0.0
Republi

In [18]:
Nigeria_data = data[data['Country/Region']== 'Nigeria']
Nigeria_data = Nigeria_data[['ObservationDate', 'Last Update', 'Confirmed', 'Deaths', 'Recovered']]
Nigeria_data

ObservationDate          Last Update  Confirmed  Deaths  Recovered
2683       02/28/2020  2020-02-28T16:23:03        1.0     0.0        0.0
2800       02/29/2020  2020-02-28T16:23:03        1.0     0.0        0.0
2926       03/01/2020  2020-02-28T16:23:03        1.0     0.0        0.0
3060       03/02/2020  2020-02-28T16:23:03        1.0     0.0        0.0
3209       03/03/2020  2020-02-28T16:23:03        1.0     0.0        0.0
3370       03/04/2020  2020-02-28T16:23:03        1.0     0.0        0.0
3538       03/05/2020  2020-02-28T16:23:03        1.0     0.0        0.0
3725       03/06/2020  2020-02-28T16:23:03        1.0     0.0        0.0
3940       03/07/2020  2020-02-28T16:23:03        1.0     0.0        0.0
4181       03/08/2020  2020-02-28T16:23:03        1.0     0.0        0.0
4410       03/09/2020  2020-03-09T09:43:03        2.0     0.0        0.0
4673       03/10/2020  2020-03-09T09:43:03        2.0     0.0        0.0
4891       03/11/2020  2020-03-09T09:43:03        2.0     0.0        0.0
5112       03/12/2020  2020-03-09T09:43:03        2.0     0.0        0.0
5243       03/13/2020  2020-03-11T20:00:00        2.0     0.0        0.0
5588       03/14/2020  2020-03-13T22:22:02        2.0     0.0        0.0
5845       03/15/2020  2020-03-13T22:22:02        2.0     0.0        0.0
6120       03/16/2020  2020-03-13T22:22:02        2.0     0.0        0.0
6390       03/17/2020  2020-03-17T12:33:06        3.0     0.0        0.0
6642       03/18/2020  2020-03-18T12:33:03        8.0     0.0        1.0
6937       03/19/2020  2020-03-18T12:33:03        8.0     0.0        1.0
7226       03/20/2020  2020-03-20T02:13:50       12.0     0.0        1.0
7507       03/21/2020  2020-03-21T15:13:46       22.0     0.0        1.0
7725       03/22/2020          3/8/20 5:31       30.0     0.0        2.0
8023       03/23/2020  2020-03-23 23:23:20       40.0     1.0        2.0
8320       03/24/2020  2020-03-24 23:41:50       44.0     1.0        2.0
8621       03/25/2020  2020-03-25 23:37:49       51.0     1.0        2.0
8924       03/26/2020  2020-03-26 23:53:24       65.0     1.0        2.0
9231       03/27/2020  2020-03-27 23:27:48       70.0     1.0        3.0
9539       03/28/2020          3/8/20 5:31       89.0     1.0        3.0
...               ...                  ...        ...     ...        ...
24571      05/14/2020  2020-05-15 02:33:02     5162.0   167.0     1180.0
24948      05/15/2020  2020-05-16 02:32:19     5450.0   171.0     1320.0
25325      05/16/2020  2020-05-17 02:32:32     5621.0   176.0     1472.0
25702      05/17/2020  2020-05-18 02:32:21     5959.0   182.0     1594.0
26079      05/18/2020  2020-05-19 02:32:18     6175.0   191.0     1644.0
26456      05/19/2020  2020-05-20 02:32:19     6401.0   192.0     1734.0
26831      05/20/2020  2020-05-21 02:32:48     6677.0   200.0     1840.0
27283      05/21/2020  2020-05-22 02:36:51     7016.0   211.0     1907.0
27735      05/22/2020  2020-05-23 02:32:42     7261.0   221.0     2007.0
28187      05/23/2020  2020-05-24 02:32:43     7526.0   221.0     2174.0
28639      05/24/2020  2020-05-25 02:32:44     7839.0   226.0     2263.0
29091      05/25/2020  2020-05-26 02:32:35     8068.0   233.0     2311.0
29543      05/26/2020  2020-05-27 02:32:31     8344.0   249.0     2385.0
29995      05/27/2020  2020-05-28 02:32:31     8733.0   254.0     2501.0
30445      05/28/2020  2020-05-29 02:32:50     8915.0   259.0     2592.0
30998      05/29/2020  2020-05-30 02:32:48     9302.0   261.0     2697.0
31552      05/30/2020  2020-05-31 02:32:45     9855.0   273.0     2856.0
32107      05/31/2020  2020-06-01 02:32:46    10162.0   287.0     3007.0
32662      06/01/2020  2020-06-02 02:33:08    10578.0   299.0     3122.0
33326      06/02/2020  2020-06-03 02:33:13    10819.0   314.0     3240.0
33989      06/03/2020  2020-06-04 02:33:14    11166.0   315.0     3329.0
34653      06/04/2020  2020-06-05 02:33:06    11516.0   323.0     3535.0
35317      06/05/2020  2020-06-06 02:33:00    11844.0   333.0   